### ISSUE TITLE
passlib takeover?

### ISSUE THREAD

**Issue by floppym** [ThirVondukr/passlib#11]
Thanks for working on this fork. I am considering using it to replace the passlib package for Gentoo Linux.

I wanted to reach out to ask if you intend to take over the passlib project on pypi at some point. Have you been in contact with Eli Collins at all?

**Comment by ThirVondukr** [comment]
Hey, I'm not sure about taking over passlib on PyPI, but I've not been in contact with Eli besides on heptapod issues.
How is passlib currently used in Gentoo? I've dropped support for python 3.8 since it recently [reached end of life](https://devguide.python.org/versions/).

**Comment by floppym** [comment]
> How is passlib currently used in Gentoo?

It's a dependency of a couple packages.

https://qa-reports.gentoo.org/output/genrdeps/rindex/dev-python/passlib

> I've dropped support for python 3.8

We don't support Python 3.8 and longer either. Currently we support 3.10 to 3.13.

https://wiki.gentoo.org/wiki/Project:Python/Implementations

**Comment by notypecheck** [comment]
@mpounsett I don't mind having someone to take a look at my past/future commits or contribute to the passlib

As for PEP541 request it may be worth trying out at this point, since Eli wasn't active at all for the past year or so, I will try reaching out to them over email first.

**Comment by chapmajs** [comment]
Looks like the Slackware SlackBuild will be switching over to this fork as well.

**Comment by lahwaacz** [comment]
Any progress? Since both `passlib` and `libpass` packages from PyPI install the same _package name_ in `site-packages`, an environment breaks if both packages get installed. For example:

```
pip install passlib
pip install libpass
pip uninstall passlib  # breaks the installation of libpass
```

Uninstall is not the only action that can break it: as soon as you start making incompatible changes in libpass, things may break one way or the other since the last installed package "takes over" the files from the first installed package.

Please solve the mess ASAP, otherwise we cannot reliably install different packages depending on `passlib` and `libpass` in the same environment.

**Comment by pbiering** [comment]
See also https://bugzilla.redhat.com/show_bug.cgi?id=2427085 (still pending), in RPM the dependency resolution looks like it's easier as there is a "Provides" option.

Imho as already stated, "passlib" needs to be taken over in PyPI or marked as deprecated or even renamed and "libpass" takes over (in case PyPI is not providing similar mechanism like RPM with "Provides") - otherwise all dependent software needs to be changed.

**Comment by lahwaacz** [comment]
I think the "Provides" option will not help as soon as `libpass` makes an incompatible API change. We would not be able to install both `passlib` and `libpass` at the same time to provide both API versions to dependent software.

**Comment by pbiering** [comment]
> I think the "Provides" option will not help as soon as `libpass` makes an incompatible API change.

Then such incompatible changes simply should not happen or software using "passlib" has simply to be adjusted.

Same can happen if "passlib" status would revert from stale to active and also introduce incompatible changes. Then  software has either to pin "passlib" to e.g. <= 1.7.4 - which potentially breaks then other software on same system requiring newer one...or  to be adjusted.

**Comment by mpounsett** [comment]
> I reached to Eli by their work email that they have on project's PyPi page, since that's the only email I was able to find. If I've read PEP correctly it's one of the requirements for the process, but I asked Eli if they could either transfer the PyPi project or participate in the development which would obviously be better.

@notypecheck : Did you hear back at all?  The project has been in limbo for over a year now (coming up on two years since the last time Eli responded to anything publicly), and I think it's time to move on taking over the PyPi project, if that isn't already in progress.  Is there anything we can help with there?

**Comment by notypecheck** [comment]
I wrote them an email back in June, but they did not respond, I did not initiate the takeover yet. As per PyPi docs I have to try reaching out to them a couple of times more, will do that today.

**Comment by mpounsett** [comment]
> As per PyPi docs I have to try reaching out to them a couple of times more

The section on [Reachability](https://peps.python.org/pep-0541/#reachability) talks about the "Package Index maintainers" making reasonable efforts to reach the project owner.  That's the PyPi maintainers, not you.   Any mention of the candidate's attempts (your attempts) to contact the project owner are less specific, using phrases like "[demonstrate their own failed attempts](https://peps.python.org/pep-0541/#continued-maintenance-of-an-abandoned-project)" or "[try to contact the current owner(s) directly](https://peps.python.org/pep-0541/#how-to-request-a-name-transfer).    

I don't see anything requiring you to attempt contact any specific number of times.

Between the emails you've sent, tags of @eli-collins in this issue, and attempts to reach Eli through the [issue on heptapod](https://foss.heptapod.net/python-libs/passlib/-/issues/187), I don't think anyone can claim that there hasn't been a significant attempt to get them to engage.

I would argue that if Eli hasn't responded within a week or so of your most recent message that you've fulfilled your obligations...   Actually, I think you met your obligations back in June.  And anyway, after you initiate a transfer request the Package Index maintainers will make their own attempts to contact Eli, so in the end that work is going to be pretty thorough.

**Comment by henning-schild** [comment]
This has been boiling for a while and the initial request from @floppym is actually coming from Gentoo Linux. With https://github.com/Kozea/Radicale moving to this fork the question of whether to fork will take over becomes more prominent.

As far as i can tell @notypecheck would potentially be willing to take over maint but @eli-collins would need to step down or approve, which seems to take its time.

Maybe it can be pointed out to PyPi that the two packages are in fact conflicting because they "own" the very same files. Knowing that might speed up the "taking ownership", and that is also the reason why distros like i.e. Gentoo Linux can not simply add `passlib` because `libpass` will have to be retired at the very same step, and there are others around who depend on `passlib` and might not be tested against `libpass`.

Not sure how to move on but it seems things are kind of stuck at the moment.

**Comment by sthen** [comment]
@henning-schild "there are others around who depend on passlib and might not be tested against libpass" - that's not really any different to a newer version of any library, I think?

An alternative would be to do what cryptodome/cryptodomex do, though as a downstream packager it would be a lot more straightforward if it could just be done as a takeover/continuation of passlib.

**Comment by lahwaacz** [comment]
> that's not really any different to a newer version of any library, I think?

That's not the point here. The issue is that installing an older and newer version of the same package, or two different packages providing the same files, simultaneously in the same environment is not possible.

**Comment by mpounsett** [comment]
> I would argue that if Eli hasn't responded within a week or so of your most recent message that you've fulfilled your obligations.

Hi again @notypecheck!  Has there been any movement in the last few weeks?

**Comment by notypecheck** [comment]
@mpounsett No, but just created a transfer request: https://github.com/pypi/support/issues/9290

**Comment by mpounsett** [comment]
Excellent news!  

I had a look at a few of the completed PEP 541 requests ... of the ones that were successful, and didn't include include any long waits for followup from the submitter, it looks like most completed in around six months.    Tiny sample, but close enough for expectation, setting I think.

**Comment by tomo-rapidspace** [comment]
> [@mpounsett](https://github.com/mpounsett) No, but just created a transfer request: [pypi/support#9290](https://github.com/pypi/support/issues/9290)

It seems the status of this request is now "Pending Transfer/Release". What does it mean? Are you able to release a new version in https://pypi.org/project/passlib/ ?

**Comment by mpounsett** [comment]
That's the stage in the process where they begin handing over the project to the new applicant.  They have queues, and I'm reasonably sure everybody involved is a volunteer, so that doesn't mean it'll be instant.    The earliest request in the `Pending Transfer/Release` state was moved there on May 1, which implies a ≥4 month wait between entering and leaving that state.

@notypecheck  Did they send you any communication when they moved to Pending last month?

I'm going to be a bit smug my "~six months" prediction on getting here was pretty close (more like five, but close enough for bragging). 😁

**Comment by notypecheck** [comment]
@mpounsett @tomo-rapidspace 
Hey, I don't think I received any emails from PyPI or was contacted in any other ways, maybe I'm missing something.  
I'm currently in a process of moving to a different country and will try get back to this when I can, if any of you could remind me of that in here that would be great.
